ateam_common/test/test_robot_assignment.cpp

```diff
@@ -0,0 +1,108 @@
+// Copyright 2021 A Team
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+// THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+#include <gtest/gtest.h>
+#include <gmock/gmock.h>
+
+#include "ateam_common/robot_assignment.hpp"
+
+using assign = ateam_common::robot_assignment::assign;
+
+TEST(RobotAssignment, assign_simple) {
+  // goals should be assigned in robot order as the robots are on top of each goal
+  std::vector<Robot> robots1 {{
+    {{1, 0, 0}, {}, {}, 0},
+    {{2, 0, 0}, {}, {}, 1},
+    {{3, 0, 0}, {}, {}, 2},
+    {{4, 0, 0}, {}, {}, 3}
+  }};
+
+  std::vector<Eigen::Vector2d> goals1 {{
+    {1, 0},
+    {2, 0},
+    {3, 0},
+    {4, 0}
+  }};
+
+  std::map<size_t, size_t> exp_result1 {{
+    {0, 0},
+    {1, 1},
+    {2, 2},
+    {3, 3},
+  }};
+  auto result1 = assign(robots1, goals1);
+  EXPECT_EQ(result1, exp_result1);
+
+
+  // less goals than robots
```

Comment:
We should break these out into different, named tests instead of having them just be commented groups in one test.

Reply:
Done

Reply:
I see you've added new tests, but those new ones don't actually test anything. It looks like they're only setting up the inputs and expected results.

Reply:
Whoops, I can't read. Added the `EXPECT_EQ` macros here that should actually implement real tests now.